### PR TITLE
Fixes fails caused by running steps on different agents #1211

### PR DESCRIPTION
**Pull request by e-schepachev**
* builder is builded on specifies agent
* cyberway image uploaded on the docker hub on a build step
* test steps try to download fresh cyberway image

Resolve #ISSUE:
-

### CODE REVIEW

File: .buildkite/steps/deploy-unit_test.sh [hunk 1]
```diff
@@ -5,12 +5,16 @@ set -euo pipefail
 docker stop mongo || true
 docker rm mongo || true
 docker volume rm cyberway-mongodb-data || true
+docker volume rm cyberway-nodeos-data || true
```
Comment (s-medvedev) [inline .buildkite/steps/deploy-unit_test.sh]: Why do you need `cyberway-nodeos-data` volume for unit-tests? As I know unit-tests runs independently without `nodeos`.

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -12,37 +14,41 @@ steps:
...
+    command: ".buildkite/steps/deploy-unit_test.sh"
+
   - wait
 
-  - label: ":cop::skin-tone-2: deploy check"
-    command: ".buildkite/steps/deploy-test.sh"
-    timeout: 20
+  - label: ":floppy_disk: upload base image"
+    command: ".buildkite/steps/publish-builder.sh"
+    branches: "master alfa develop ci-* v*.*.*"
```
Comment (s-medvedev) [inline .buildkite/pipeline.yml]: This step duplicates in line 10-12

File: .buildkite/steps/deploy-unit_test.sh (hunk 1 above)
Comment (s-medvedev) [reply .buildkite/steps/deploy-unit_test.sh]: Ok. These need for correct execution when `deploy-unit_test.sh` run on new agent.

**Comment by s-medvedev** [comment]
What happens if the build and publish step is performed on different agents. It seems to me that the publication will not succeed, since the `docker tag` will fail with missing image.

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -1,18 +1,35 @@
...
+      - ".buildkite/steps/upload-image.sh"
 
   - wait
 
-  - label: ":docker: build docker image"
-    command: ".buildkite/steps/build-image.sh"
+  - label: ":cop::skin-tone-2: deploy check"
+    command: ".buildkite/steps/deploy-test.sh"
+    timeout: 20
+
```
Comment (muhazzz) [inline .buildkite/pipeline.yml]: Need "wait" here

File: .buildkite/pipeline.yml [hunk 4]
```diff
@@ -1,18 +1,35 @@
...
+  - label: ":cop::skin-tone-2: api test"
+    command: ".buildkite/steps/api-test.sh"
+    timeout: 20
+
```
Comment (muhazzz) [inline .buildkite/pipeline.yml]: Need "wait"

**Pull request merged**